src/Factory/EventManagerFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zend-eventmanager for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+namespace Zend\EventManager\Factory;
+
+use Zend\EventManager\EventManager;
+use Zend\EventManager\ListenerPluginManager;
+use Zend\ServiceManager\FactoryInterface;
+
+class EventManagerFactory implements FactoryInterface
+{
+    /**
+     * Override config key in an extending class to point to more specific key if needed
+     *
+     * @var string
+     */
+    protected $configKey = 'event_manager';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __invoke(ServiceLocatorInterface $serviceLocator, $requestedName, array $options = [])
+    {
+        // @TODO: ZF3 should have a Config accessible through FQCN
+        $config = $serviceLocator->get('Config');
+        $config = isset($config[$this->configKey]) ? $config[$this->configKey] : [];
+
+        $eventManager = new EventManager($serviceLocator->get(ListenerPluginManager::class));
```

Comment:
Actually this is wrong but I'll fix that tomorrow. The instantiator is not a plugin manager. The instantiator is just a callable that accepts a name.